File: 2626_array_reduce_transformation/solution.rb
# LeetCode 2626 - Array Reduce Transformation
# https://leetcode.com/problems/array-reduce-transformation/

# @param {Integer[]} nums
# @param {Proc} fn
# @param {Object} init
# @return {Object}
def reduce(nums, fn, init)
  acc = init
  nums.each { |x| acc = fn.call(acc, x) }
  acc
end

File: 2626_array_reduce_transformation/solution.ts
// LeetCode 2626 - Array Reduce Transformation
// https://leetcode.com/problems/array-reduce-transformation/

export function reduce(nums: any, fn: any, init: any): any {
    let acc = init;
    for (const x of nums) acc = fn(acc, x);
    return acc;
}
